impl_core/src/main/java/io/opencensus/implcore/trace/SpanImpl.java

```diff
@@ -507,6 +516,8 @@ private AttributesWithCapacity(int capacity) {
     // Users must call this method instead of put to keep count of the total number of entries
     // inserted.
     private void putAttribute(String key, AttributeValue value) {
+      Utils.checkNotNull(key, "key");
```

Comment:
I don't think that the null checks in private methods are necessary, since the arguments should already have been validated in the public methods.

Reply:
Agree completely. I avoided privates and deprecated methods but missed this one!

Reply:
Did you remove the wrong null check?  The null checks in the two private methods are still there.

Reply:
Ouf. Sorry about that. I could sworn I changed all of them but definitely missed these two. I just force pushed some changes that should rectify that issue!